pymodbus/transport/transport.py

```diff
@@ -176,39 +176,37 @@ def __init__(
         else:
             host = self.comm_params.host
             port = int(self.comm_params.port)
-        if self.comm_params.comm_type == CommType.SERIAL:
-            host, port = self.init_setup_serial(host, port)
-            if not host and not port:
-                return
-        if host == NULLMODEM_HOST:
+        if NULLMODEM_HOST in host:
```

Comment:
this allows "NULLMODEM_HOST:7109" for all commtype...and it is only defined for serial.

Not sure it is a big problem, but at least a concern.

Reply:
This is functionally possible in the old/existing code as well:
https://github.com/pymodbus-dev/pymodbus/blob/6b1ee62b34c50f4e8f3f8c9a9b42e02ff2f8ffe4/pymodbus/transport/transport.py#L183-L185

It's certainly possible that was unintended, in which case the refactor exposed a bug :)

Reply:
no the old code checked for exact match, those lines would not accept "NULLMODEM:9657"

Reply:
d'oh, you're right